api/v1alpha1/networkfence_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FencingOperationResult string
+
+const (
+	// OperationResultSucceeded represents the Succeeded operation state.
+	FencingOperationResultSucceeded FencingOperationResult = "Succeeded"
+
+	// OperationResultFailed represents the Failed operation state.
+	FencingOperationResultFailed FencingOperationResult = "Failed"
+)
+
+// NetworkFenceSpec defines the desired state of NetworkFence
+type NetworkFenceSpec struct {
+	// Operation specifes whether fence or unfence operation is to be performed
+	// on the corresponding CIDR blocks.
+	// It's can be set to either "fence" or "unfence".
+	Operation string `json:"operation,omitempty"`
```

Comment:
This would also be something I would expect to happen. Create an instance of the `NetworkFence` CR to execute the underlaying fence operation. When the CR is deleted the unfence operation would be executed.

Can the CIDR list be modified after the initial creation of the CR? I guess that would be practical, so that a single CSI-node can be fenced/unfenced by editing the CR, while keeping the state of others in tact.

Reply:
Having an update operation for the CR may cause issues during the unfence operation; as we perform unfence on a CIDR when the `NetworkFence` CR corresponding to it is deleted.
1. Assuming we have a CR with CIDR `10.23.53.42/24` , this will result in network fencing of `10.23.53.42/24`
2. Now, if an update is done on the CR, to say `10.15.35.64/24`, this will now fence `10.15.35.64/24`.
3. On deletion of the CR now, only `10.15.35.64/24` will be unfenced and the first CIDR(`10.23.53.42/24` ) will still remain fenced.

wdyt? @nixpanic @Madhu-1